solutions/USACO dec19-milkvisits-silver.mdx

```diff
@@ -2,13 +2,44 @@
 id: usaco-968
 source: USACO Silver 2019 December
 title: Milk Visits
-author: Qi Wang
+author: Qi Wang, Tanish Tyagi
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_milkvisits_silver_dec19.html)
 
 **Time Complexity:** $\mathcal{O}(N + M)$
 
+We can use a DFS to create connected components from an adjacency list (which is made when we process the roads between the farms). Each connected component will contain farms that all have the same breed. 
+
+Adjacency List based off sample test case: <br/>
+$1$ $\rightarrow$ $2$, $5$  <br/>
+$2$ $\rightarrow$ $1$, $3$, $4$  <br/>
+$3$ $\rightarrow$ $2$ <br/>
+$4$ $\rightarrow$ $2$ <br/>
+$5$ $\rightarrow$ $1$ <br/>
+
+There are three connected components of the same cow breed in this graph:
+- Component $1$: Farms $1$, $2$, and $4$
+- Component $2$: Farm $3$
+- Component $3$: Farm $5$
+
+Create an array $\texttt{comp}$ such that $\texttt{comp}[A]$ is the component of farm $A$.
+
+$\texttt{comp} = [1, 1, 2, 1, 3]$ (1-indexed)
+
+We evaluate 3 conditions to check whether the farmer will be happy:
+
+1. Farm $A$ and $B$ are part of the same component.  This means the path between $A$ and $B$ only contains cows of the same breed. If the farmer prefers milk from this breed, then we should output $1$.
+2. Farm $A$ and $B$ are part of the different components. This means that the farmer will always be satisfied because the path between $A$ and $B$ contains both breeds of cows. We should output $1$.
```

Comment:
```suggestion
2. Farms $A$ and $B$ are part of different components. This means that the farmer will always be satisfied because the path between $A$ and $B$ contains both breeds of cows. We should output $1$.
```